r/R/utils.R

```diff
@@ -45,8 +45,11 @@ document_granges <- function(
   ), intro_noun, bullets)
 }
 
-# Function which prints a message using shell echo
-# Useful for printing messages from inside mclapply when running in Rstudio
-message_parallel <- function(msg){
+# Function which prints a message using shell echo.
+# Useful for printing messages from inside mclapply when running in Rstudio.
+log_progress <- function(msg, add_timestamp = TRUE){
+  if (add_timestamp) {
+    msg <- paste0(format(Sys.time(), "%Y-%m-%d %H:%M:%S "), msg)
+  }
   system(sprintf('echo "%s"', paste0(msg, collapse="")))
```

Comment:
I just did a bit of extra research -- looks like it is possible to explicitly detect running in RStudio via `.Platform$GUI == "RStudio"`. This might allow use `message()` to play a bit more nicely with R's normal systems while just resorting to `system('echo')` on RStudio

Reply:
Interesting, let me put in some better handling for this